project-base/storefront/components/Pages/Order/TransportAndPayment/TransportAndPaymentContent.tsx

```diff
@@ -120,58 +67,52 @@ export const TransportAndPaymentContent: FC<TransportAndPaymentContentProps> = (
         router.push(contactInformationUrl);
     };
 
-    const lastOrderPickupPlace: ListedStoreFragmentApi | null = useMemo(() => {
-        if (lastOrder?.pickupPlaceIdentifier === undefined || lastOrder.pickupPlaceIdentifier === null) {
-            return null;
-        }
-
-        if (packeteryPickupPoint?.identifier === lastOrder.pickupPlaceIdentifier) {
-            return packeteryPickupPoint;
-        }
-
-        if (pickupPlaceData?.store !== undefined && pickupPlaceData.store !== null) {
-            return getGtmPickupPlaceFromStore(lastOrder.pickupPlaceIdentifier, pickupPlaceData.store);
-        }
-
-        return getGtmPickupPlaceFromLastOrder(lastOrder.pickupPlaceIdentifier, lastOrder);
-    }, [lastOrder, packeteryPickupPoint, pickupPlaceData?.store]);
-
     return (
         <>
-            {transports !== undefined && transports.length > 0 && (
-                <TransportAndPaymentSelect
-                    transports={transports}
-                    lastOrderPickupPlace={lastOrderPickupPlace}
-                    lastOrderTransportUuid={lastOrder?.transport.uuid ?? null}
-                    lastOrderPaymentUuid={lastOrder?.payment.uuid ?? null}
-                    changeTransportInCart={changeTransportInCart}
-                    changePaymentInCart={changePaymentInCart}
-                    isTransportSelectionLoading={isTransportSelectionLoading}
-                />
-            )}
-            <OrderAction
-                buttonBack={t('Back')}
-                buttonNext={t('Contact information')}
-                hasDisabledLook={
-                    hasValidationErrors(transportAndPaymentValidationMessages) ||
-                    isTransportSelectionLoading ||
-                    isPaymentSelectionLoading
-                }
-                isLoading={
-                    (isTransportSelectionLoading || isPaymentSelectionLoading) && transport !== null && payment !== null
-                }
-                withGapTop
-                withGapBottom
-                buttonBackLink={cartUrl}
-                nextStepClickHandler={onSelectTransportAndPaymentHandler}
-            />
-            {isErrorPopupVisible && (
-                <ErrorPopup
-                    onCloseCallback={() => setErrorPopupVisibility(false)}
-                    fields={transportAndPaymentValidationMessages}
-                    gtmMessageOrigin={GtmMessageOriginType.transport_and_payment_page}
-                />
-            )}
+            <OrderLayout
+                activeStep={2}
+                isTransportOrPaymentLoading={isTransportSelectionLoading || isPaymentSelectionLoading}
+                isLoading={isTransportAndPaymentSkeletonVisible}
+            >
+                {isTransportAndPaymentSkeletonVisible ? (
+                    <Skeleton className="h-64 w-full" />
+                ) : (
```

Comment:
I still think this can be improved. What about this?

```suggestion
            <OrderLayout
                activeStep={2}
                isTransportOrPaymentLoading={isTransportSelectionLoading || isPaymentSelectionLoading}
                isLoading={isTransportAndPaymentSkeletonVisible}
                skeleton={<Skeleton className="h-64 w-full" />}
            >
               // content itself
```

And then adjust proper Skeleton there

Reply:
Though it's not the way I would do it (I believe a skeleton is a content as any other) I tried to follow your suggeestions. I also removed the `isLoading` prop and only base my decision on the `contentSkeleton` prop (if its there, I show it, otherwise I dont)

Reply:
Nice, I think since you have the skeleton for the page in OrderLayout, the content skeleton can be part of this main skeleton. Especially when it makes the code cleaner.